src/actions/eventActions.js

```diff
@@ -14,24 +14,24 @@ export const POST_EVENT_TRY = 'POST_EVENT_TRY'
 export const POST_EVENT_SUCCEED = 'POST_EVENT_SUCCEED'
 export const POST_EVENT_FAIL = 'POST_EVENT_FAIL'
 export const ADD_EVENT = 'ADD_EVENT'
+export const CLEAR_EVENTS = 'CLEAR_EVENTS'
 
 export const pagination = paginationActions(EVENTS)
 
 export const linksHeaderName = 'links'
 
+export const clearEvents = () => ({
+  type: CLEAR_EVENTS
+})
+
 export const fetchEvent = (incidentId, eventId) => reduxBackedPromise(
-    [getEventsEndPoint(incidentId) + eventId],
-    getEventActionSet(incidentId, eventId)
+  [getEventsEndPoint(incidentId) + eventId],
+  getEventActionSet(incidentId, eventId)
 )
 
 export const fetchEvents = (filter) => reduxBackedPromise(
-        getEventsFetchArgs(filter),
-        getEventsActionSet(filter.incidentId)
-    )
-
-export const fetchUncorrelatedEvents = (filter) => reduxBackedPromise(
-        ['events/' + filterService.serializeFiltersForUrl(filter)],
-        getEventsActionSet(null)
```

Comment:
Did you merged this step into fetchEvents?

Reply:
Yes, getEventsFetchArgs will call getEventsEndPoint to generate the base url (/incidents/X/events/ when incidentId has value X or /events/ when incidentId has no value)